statistics/update.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/util/sqlexec"
+)
+
+type tableDeltaMap map[int64]*variable.TableDelta
+
+func (m tableDeltaMap) update(id int64, delta int64, count int64) {
+	item, ok := m[id]
+	if !ok {
+		m[id] = &variable.TableDelta{Delta: delta, Count: count}
+	} else {
+		item.Delta += delta
+		item.Count += count
+	}
+}
+
+func (m tableDeltaMap) merge(handle *SessionStatsCollector) {
+	handle.Lock()
+	defer handle.Unlock()
+	for id, item := range handle.mapper {
+		m.update(id, item.Delta, item.Count)
+	}
+	handle.mapper = make(tableDeltaMap)
+}
+
+// SessionStatsCollector is a list item that holds the update mapper. If you want write or read mapper, you must add the lock.
+type SessionStatsCollector struct {
+	sync.Mutex
+
+	mapper tableDeltaMap
+	prev   *SessionStatsCollector
+	next   *SessionStatsCollector
+}
+
+// Update will updates the delta and count for one table id.
+func (h *SessionStatsCollector) Update(id int64, delta int64, count int64) {
+	h.Lock()
+	defer h.Unlock()
+	h.mapper.update(id, delta, count)
+}
+
+// updateManager management the delta information in tidb. It collects the delta map from all sessions and write them to tikv.
+type updateManager struct {
```

Comment:
Do we really need `updateManager`, how about implement the methods in `Handle` directly?